tests/ord-service/tests/subscription_flow_test.go

```diff
@@ -0,0 +1,184 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tests
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/tidwall/sjson"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/kyma-incubator/compass/tests/pkg/ptr"
+	"github.com/kyma-incubator/compass/tests/pkg/tenant"
+	"github.com/kyma-incubator/compass/tests/pkg/token"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	tenantPathParamValue       = "tenant"
+	regionPathParamValue       = "eu-1"
+	defaultSubdomain           = "default-subdomain"
+)
+
+type Tenant struct {
+	TenantID               string
+	SubaccountID           string
+	CustomerID             string
+	Subdomain              string
+	SubscriptionProviderID string
+}
+
+func TestSubscriptionFlow(t *testing.T) {
+	ctx := context.Background()
+	defaultTenantId := tenant.TestTenants.GetDefaultTenantID()
+	secondaryTenant := tenant.TestTenants.GetIDByName(t, tenant.ApplicationsForRuntimeTenantName)
+	subscriptionProviderID := "xs-app-name"
+	subscriptionConsumerID := "1f538f34-30bf-4d3d-aeaa-02e69eef84ae"
+	accountID := uuid.New().String()
+
+	runtimeInput := graphql.RuntimeInput{
+		Name:        "testingRuntime",
+		Description: ptr.String("testingRuntime-description"),
+		Labels:      graphql.Labels{testConfig.SubscriptionProviderLabelKey: subscriptionProviderID, "region": regionPathParamValue},
+	}
+
+	runtime, err := fixtures.RegisterRuntimeFromInputWithinTenant(t, ctx, dexGraphQLClient, defaultTenantId, &runtimeInput)
+	defer fixtures.CleanupRuntime(t, ctx, dexGraphQLClient, defaultTenantId, &runtime)
+	require.NoError(t, err)
+	require.NotEmpty(t, runtime.ID)
+
+	app, err := fixtures.RegisterApplication(t, ctx, dexGraphQLClient, "testingApp", secondaryTenant)
+	defer fixtures.CleanupApplication(t, ctx, dexGraphQLClient, secondaryTenant, &app)
+	require.NoError(t, err)
+	require.NotEmpty(t, app.ID)
+
+	consumerApp, err := fixtures.RegisterApplication(t, ctx, dexGraphQLClient, "consumerApp", secondaryTenant)
+	defer fixtures.CleanupApplication(t, ctx, dexGraphQLClient, secondaryTenant, &consumerApp)
+	require.NoError(t, err)
+	require.NotEmpty(t, consumerApp.ID)
+
+	// Create label definition
+	scenarios := []string{"DEFAULT", "consumer-test-scenario"}
+	fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, secondaryTenant, scenarios)
+	defer fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, secondaryTenant, scenarios[:1])
+
+	// Create automatic scenario assigment for consumer subaccount
+	asaInput := fixtures.FixAutomaticScenarioAssigmentInput(scenariosLabel, selectorKey, subscriptionConsumerID)
```

Comment:
The first argument here should not be `scenariosLabel` but should be `scenarios[1]` or `consumer-test-scenario`